src/screens/nano/NanoContractSelectBlueprint.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef, useState } from 'react';
+import { t } from 'ttag'
+import BackButton from '../../components/BackButton';
+import colors from '../../index.module.scss';
+import hathorLib from '@hathor/wallet-lib';
+import ReactLoading from 'react-loading';
+import { useNavigate } from 'react-router-dom';
+
+/**
+ * Select blueprint, fetch information and show it
+ *
+ * @memberof Screens
+ */
+function NanoContractSelectBlueprint() {
+
+  const blueprintIdRef = useRef(null);
+  const formSelectBlueprintRef = useRef(null);
+
+  const navigate = useNavigate();
+
+  // fetching {boolean} If it's fetching blueprint information data, to show loading
+  const [fetching, setFetching] = useState(false);
+  // blueprintInformation {Object} Blueprint information data from full node API
+  const [blueprintInformation, setBlueprintInformation] = useState(null);
+  // errorMessage {string} Message to show when error happens on the form
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const onSelectBlueprint = async () => {
+    setErrorMessage('');
+    const isValid = formSelectBlueprintRef.current.checkValidity();
+    
+    if (!isValid) {
+      formSelectBlueprintRef.current.classList.add('was-validated')
+      return;
+    }
+
+    const blueprintId = blueprintIdRef.current.value;
+    try {
+      setFetching(true);
+      const data = await hathorLib.ncApi.getBlueprintInformation(blueprintId);
+      setBlueprintInformation(data);
+    } catch (e) {
+      setErrorMessage(e.message);
+    } finally {
+      setFetching(false);
+    }
+  }
+
+  const onSelectNewBlueprint = () => {
+    blueprintIdRef.current.value = '';
+    setBlueprintInformation(null);
+  }
+
+  const onConfirmBlueprintSelection = () => {
+    navigate('/nano_contract/execute_method/', {
+      state: {
+        blueprintInformation,
+        method: 'initialize'
+      },
+    });
+  }
+
+  const renderButtons = () => {
+    if (!blueprintInformation) {
+      return (
+        <button type="button" className="btn btn-hathor mr-3" onClick={onSelectBlueprint}>{t`Confirm`}</button>
```

Comment:
suggestion(non-blocking): Rename this button to "Fetch blueprint data", "Search" or other label that conveys a query with no side effects instead of a confirmation.

"Confirm" is the same name used when the blueprint is selected and we are trigerring the contract initialization.

```suggestion
        <button type="button" className="btn btn-hathor mr-3" onClick={onSelectBlueprint}>{t`Fetch blueprint data`}</button>
```

Reply:
Much better, thanks!

<img width="1305" alt="image" src="https://github.com/HathorNetwork/hathor-wallet/assets/3298774/fe6b26a5-19ac-44bf-91af-0489a9443454">